sycl/source/detail/platform_impl.cpp

```diff
@@ -116,8 +116,9 @@ vector_class<platform> platform_impl::get_platforms() {
         platform Platform = detail::createSyclObjFromImpl<platform>(
             getOrMakePlatformImpl(PiPlatform, Plugins[i]));
         // Skip platforms which do not contain requested device types
-        if (!Platform.get_devices(ForcedType).empty() &&
-            !IsBannedPlatform(Platform))
+        if (!IsBannedPlatform(Platform) &&
+            (ForcedType == info::device_type::all ||
```

Comment:
Why have you added this? It is a change in current behavior that I am not sure we want and is also unrelated to the rest of the change, right?

Reply:
The original condition has two parts to include the given platform into get_platforms().
First, check if the platform includes any device of the forced type.
Second, check if the platform is banned.

For the first condition, if there is no SYCL_DEVICE_TYPE is set, we use ForcedType==device_type::all, which means the platform will contain a device for sure.
So, it is enough to just check if ForcedType is device_type::all.
The second condition still filter the platform if the Platform is banned platform.

By doing this change, we can avoid calling ::get_devices() everytime get_platform() is called.